src/com/facebook/buck/jvm/java/abi/StubJarClassEntry.java

```diff
@@ -106,7 +106,7 @@ public static StubJarClassEntry of(
 
     // The synthetic package-info class is how package annotations are recorded; that one is
     // actually used by the compiler
-    if (!isAnonymousOrLocalOrSyntheticClass(stub)
+    if (!((isSyntheticClass(stub) && !isKotlinModule) || isAnonymousOrLocalClass(stub))
```

Comment:
If we can simplify this logic and match formatting, that'd be great.

Reply:
Simplified.